lib/fs.js

```diff
@@ -809,13 +832,22 @@ Object.defineProperty(fs.write, internalUtil.customPromisifyArgs,
 // OR
 //  fs.writeSync(fd, string[, position[, encoding]]);
 fs.writeSync = function(fd, buffer, offset, length, position) {
+  if (!Number.isInteger(fd))
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'fd', 'number');
+  if (fd < 0 || fd > 0xFFFFFFFF)
+    throw new errors.RangeError('ERR_OUT_OF_RANGE', 'fd');
   if (isUint8Array(buffer)) {
     if (position === undefined)
       position = null;
     if (typeof offset !== 'number')
       offset = 0;
     if (typeof length !== 'number')
       length = buffer.length - offset;
+    const byteLength = buffer.byteLength;
+    if (offset > byteLength)
+      throw new errors.RangeError('ERR_OUT_OF_RANGE', 'offset');
```

Comment:
`ERR_OUT_OF_RANGE` is reduced into `ERR_VALUE_OUT_OF_RANGE` in https://github.com/nodejs/node/pull/17648. So here would be better to use `ERR_VALUE_OUT_OF_RANGE`.

Reply:
+1

Reply:
@jasnell ^

Reply:
yep, was holding off a bit to see if/when #17648 was going to land.

Reply:
oh.. right, and the latest updates remove the out of range errors anyway so this is no longer relevant ;-)